deploy/0_deploy.ts

```diff
@@ -41,82 +62,176 @@ const ensPublicResolver: PublicResolver = hre.artifacts.require(
+        (await iexecAccessorsInstance.eip712domain_separator()) != ethers.constants.HashZero;
+    if (!iexecInitialized) {
+        await IexecMaintenanceDelegate__factory.connect(erc1538ProxyAddress, owner)
+            .configure(
+                rlcInstanceAddress,
+                'Staked RLC',
+                'SRLC',
+                9, // TODO: generic ?
+                appRegistryAddress,
+                datasetRegistryAddress,
+                workerpoolRegistryAddress,
+                ethers.constants.AddressZero,
+            )
+            .then((tx) => tx.wait());
+    }
+    // Set categories
+    const catCountBefore = await iexecAccessorsInstance.countCategory();
+    const categories = CONFIG.categories as Category[];
+    for (let i = catCountBefore.toNumber(); i < categories.length; i++) {
+        const category = categories[i];
+        await IexecCategoryManager__factory.connect(erc1538ProxyAddress, owner).createCategory(
+            category.name,
+            JSON.stringify(category.description),
+            category.workClockTimeRef,
+        );
+    }
+    const catCountAfter = await iexecAccessorsInstance.countCategory();
+    console.log(`countCategory is now: ${catCountAfter}`);
+    for (let i = 0; i < catCountAfter.toNumber(); i++) {
+        console.log(`Category ${i}: ${await iexecAccessorsInstance.viewCategory(i)}`);
+    }
 };
 
+/**
+ * Extract base contract name from contract factory name.
+ * Inputting `MyBoxContract__factory` returns `MyBoxContract`.
+ */
+function getBaseNameFromContractFactory(contractFactory: any) {
+    const name = contractFactory.constructor.name;
+    return name.replace('__factory', '');
+}
+
+/**
+ * Deploy through a GenericFactory a contract [and optionally trigger call]
+ */
+async function deploy(contractFactory: ContractFactory, constructorArgs?: any[], call?: string) {
+    let bytecode = constructorArgs
+        ? contractFactory.getDeployTransaction(...constructorArgs).data
+        : contractFactory.getDeployTransaction().data;
+    if (!bytecode) {
+        throw new Error('Failed to prepare bytecode');
+    }
+    let contractAddress = call
+        ? await genericFactoryInstance.predictAddressWithCall(bytecode, salt, call)
+        : await genericFactoryInstance.predictAddress(bytecode, salt);
+    const previouslyDeployed = (await ethers.provider.getCode(contractAddress)) !== '0x';
+    if (!previouslyDeployed) {
+        call
+            ? await genericFactoryInstance
+                  .createContractAndCall(bytecode, salt, call)
+                  .then((tx) => tx.wait())
+            : await genericFactoryInstance.createContract(bytecode, salt).then((tx) => tx.wait());
+    }
+    const contractName = getBaseNameFromContractFactory(contractFactory);
+    console.log(
+        `${contractName}: ${contractAddress} ${previouslyDeployed ? ' (previously deployed)' : ''}`,
+    );
+    await deployments.save(contractName, {
+        abi: [],
```

Comment:
Can we use `contractFactory.constructor.abi` here?
```suggestion
        abi: contractFactory.constructor.abi,
```

Reply:
Though the same, we don't use it, but let's add it :) [Update wording and add abi on save deployment](https://github.com/iExecBlockchainComputing/PoCo/pull/58/commits/c0df6b1d4f10f8dcf16f699947f12c40a46c3155)

Reply:
If it's not used we can remove it. It's up to you.